typescript/sdk/src/providers/ProviderType.ts

```diff
@@ -48,6 +48,46 @@ export const PROTOCOL_TO_DEFAULT_PROVIDER_TYPE: Record<
 
 export type ProviderMap<Value> = Partial<Record<ProviderType, Value>>;
 
+type ProtocolTypesMapping = {
+  [ProtocolType.Ethereum]: {
+    transaction: EthersV5Transaction;
+    provider: EthersV5Provider;
+    contract: EthersV5Contract;
+    receipt: EthersV5TransactionReceipt;
+  };
+  [ProtocolType.Sealevel]: {
+    transaction: SolanaWeb3Transaction;
+    provider: SolanaWeb3Provider;
+    contract: SolanaWeb3Contract;
+    receipt: SolanaWeb3TransactionReceipt;
+  };
+  [ProtocolType.Cosmos]: {
+    transaction: CosmJsWasmTransaction;
+    provider: CosmJsWasmProvider;
+    contract: CosmJsWasmContract;
+    receipt: CosmJsWasmTransactionReceipt;
+  };
+};
+export type SupportedProtocolType = keyof ProtocolTypesMapping;
```

Comment:
I'd be more in favor of removing the protocol types we don't actually support instead of creating new types and mappings for a subset of them.